library/core/src/slice/iter.rs

```diff
@@ -190,6 +191,7 @@ impl<T> AsRef<[T]> for Iter<'_, T> {
 /// [`iter_mut`]: slice::iter_mut
 /// [slices]: slice
 #[stable(feature = "rust1", since = "1.0.0")]
+#[repr(C)] // *Not* a guarantee, but keeps the codegen tests consistent
```

Comment:
Is this about layout randomization? If so then using the `//@ needs-deterministic-layouts` test annotation should be better, that way people can still get randomized iters.

Reply:
`#[cfg_attr(not(doc), repr(C))]` is also used in some places.